### DISCUSSION TITLE
dispose uv1 ?

### DISCUSSION THREAD

**Discussion by echadwick-artist** [donmccurdy/glTF-Transform#610] [Q&A] [answered]
Hi Don et.al.,

Is it possible to remove TEXCOORD_1 from glTF models using glTF-Transform?

I suppose it would be similar to disposing of vertex colors? https://github.com/donmccurdy/glTF-Transform/discussions/309

The reasoning is to strip out the unused 2nd uv to solve unwanted three.js behavior with our models, see https://github.com/mrdoob/three.js/issues/24325

Thanks!

**Comment by donmccurdy** [comment, marked as answer]
This could be done a bit more simply if you know exactly which textures and extensions are (or are not) used by the models you're processing, but here's a conservative solution to:

1. reassign all material texture slots to use texCoord=0
2. remove all `TEXCOORD_N` vertex attributes for N ≥ 1
3. dispose of associated accessors

```js
import { TextureInfo } from '@gltf-transform/core';

for (const graphEdge of document.getGraph().listEdges()) {
	const graphChild = graphEdge.getChild();
	if (graphChild instanceof TextureInfo) {
		graphChild.setTexCoord(0);
	}
}

for (const mesh of document.getRoot().listMeshes()) {
	for (const prim of mesh.listPrimitives()) {
		for (let i = 1; !!prim.getAttribute(`TEXCOORD_${i}`); i++) {
			prim.getAttribute(`TEXCOORD_${i}`).dispose();
		}
	}
}
```

This could be used in a script with node.js, or run in the script tab at https://gltf.report/ —

![Screen Shot 2022-07-07 at 10 52 54 PM](https://user-images.githubusercontent.com/1848368/177907610-062a27ca-4bbc-477b-b2b6-8b90843a1dbf.png)
